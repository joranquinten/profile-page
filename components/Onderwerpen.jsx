import React from 'react';
import renderHTML from 'react-render-html';
import styled from 'styled-components';

import { Tag, Card } from 'antd';

import { sorterByProperty } from '../common/helpers';
import { topics } from '../data/nl';

const typeColorMap = {
  frameworks: 'volcano',
  languages: 'gold',
  methodologies: 'purple',
  concepts: 'cyan',
  platforms: 'geekblue',
  other: 'blue',
};

const StyledOnderwerpen = styled.div`
  .ant-tag {
    margin: 0.4rem !important;
    display: inline-block;
  }
`;

const Onderwerpen = () => (
  <StyledOnderwerpen>
    <Card title="Onderwerpen">
      {renderHTML(topics.description)}
      {topics.list.sort(sorterByProperty('name')).map(item => (
        <Tag key={item.name} color={typeColorMap[item.type]}>
          {item.name}
        </Tag>
      ))}
    </Card>
  </StyledOnderwerpen>
);

export default Onderwerpen;
