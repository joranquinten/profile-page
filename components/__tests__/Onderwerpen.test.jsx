import { shallow } from 'enzyme';

import Onderwerpen from '../Onderwerpen';

jest.mock('../../data/nl.js');

describe('<Onderwerpen />', () => {
  it('should render without crashing', () => {
    expect(shallow(<Onderwerpen />)).toMatchSnapshot();
  });
});
