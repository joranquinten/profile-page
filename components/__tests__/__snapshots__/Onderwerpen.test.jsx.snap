// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Onderwerpen /> should render without crashing 1`] = `
<Onderwerpen__StyledOnderwerpen>
  <Card
    title="Onderwerpen"
  >
    <p
      key="0"
    >
      topicsDescription
    </p>
    <Tag
      closable={false}
      color="volcano"
      key="topicName1"
    >
      topicName1
    </Tag>
    <Tag
      closable={false}
      color="volcano"
      key="topicName2"
    >
      topicName2
    </Tag>
    <Tag
      closable={false}
      color="gold"
      key="topicName3"
    >
      topicName3
    </Tag>
    <Tag
      closable={false}
      color="purple"
      key="topicName4"
    >
      topicName4
    </Tag>
    <Tag
      closable={false}
      color="blue"
      key="topicName5"
    >
      topicName5
    </Tag>
    <Tag
      closable={false}
      color="cyan"
      key="topicName6"
    >
      topicName6
    </Tag>
    <Tag
      closable={false}
      color="geekblue"
      key="topicName7"
    >
      topicName7
    </Tag>
  </Card>
</Onderwerpen__StyledOnderwerpen>
`;
